From afc861c7231de431d852d33f80f196bee654c5ba Mon Sep 17 00:00:00 2001
From: Kristian Amlie <kristian.amlie@cfengine.com>
Date: Tue, 7 Jan 2014 08:03:48 +0100
Subject: [PATCH] Tiny optimization in safe_open.

We already know that next_component is neither null nor empty in the
second branch, because of 7f137f0ad5.

(cherry picked from commit e890b4a420a9d7857c8a6a89a5aa5db061045faf)
---
 libutils/file_lib.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/libutils/file_lib.c b/libutils/file_lib.c
index f111785..436f3b3 100644
--- a/libutils/file_lib.c
+++ b/libutils/file_lib.c
@@ -152,15 +152,15 @@ int safe_open(const char *pathname, int flags, ...)
         first_dir = "/";
         // Eliminate double slashes.
         while (*(++next_component) == '/') { /*noop*/ }
+        if (!*next_component)
+        {
+            next_component = NULL;
+        }
     }
     else
     {
         first_dir = ".";
     }
-    if (!*next_component)
-    {
-        next_component = NULL;
-    }
     currentfd = open(first_dir, O_RDONLY);
     if (currentfd < 0)
     {
-- 
1.8.1.5

